ci-bench-runner/src/bencher_dev.rs

```diff
@@ -81,32 +87,33 @@ impl BencherDev {
     }
 }
 
-/// Converts the instruction counts map into Bencher Metric Format (BMF)
+/// Converts the instruction counts map into Bencher Metric Format (BMF) and appends them to the map
 ///
 /// See <https://bencher.dev/docs/explanation/adapters#-json> for details
-fn icounts_to_bmf(icounts: HashMap<String, f64>) -> JsonResultsMap {
-    let instructions: MetricKind = INSTRUCTIONS_SLUG_STR.parse().unwrap();
-    icounts
-        .into_iter()
-        .filter_map(|(scenario_name, value)| {
-            let benchmark_name = match scenario_name.parse() {
-                Ok(name) => name,
-                Err(_) => {
-                    error!(
-                        scenario_name,
-                        "benchmark name does not conform to bencher.dev's requirements, ignoring"
-                    );
-                    return None;
-                }
-            };
+fn append_results_as_bmf(
```

Comment:
This might be more of a style thing but I feel like my inclination would be to try and avoid passing in mutable state by having this function do the conversion of `HashMap<String, f64>`, `MetricKind` to BMF and return `JsonMetricsMap`, leaving pushing the BMF into the `JsonResultsMap` to the callers. Is there a reason to avoid that I'm overlooking?